pkg/apis/monitoring/v1/types.go

```diff
@@ -150,6 +150,77 @@ func (pc *ProxyConfig) Validate() error {
 	return nil
 }
 
+type SafeHTTPHeader struct {
+	// Header values.
+	//
+	// +kubebuilder:validation:MinItems=1
+	// +kubebuilder:validation:items:MinLength=1
+	// +listType=set
+	// +optional
+	Values []string `json:"values,omitempty"`
+	// Headers values. Hidden in configuration page.
+	//
+	// +kubebuilder:validation:MinItems=1
+	// +listType=set
+	// +optional
+	Secrets []v1.SecretKeySelector `json:"secrets,omitempty"`
+}
+
+// Validate semantically validates the given SafeHttpHeader.
+func (c *SafeHTTPHeader) Validate() error {
+	if c == nil {
+		return nil
+	}
+
+	if len(c.Secrets) == 0 {
+		return errors.New("Secrets selectors must not be empty")
+	}
+
+	for _, v := range c.Secrets {
+		if v == (v1.SecretKeySelector{}) {
+			return errors.New("Secrets selector must be defined")
+		}
+	}
+
+	return nil
+}
+
+type HTTPHeader struct {
+	SafeHTTPHeader `json:",inline"`
+
+	// Files to read header values from.
+	//
+	// +kubebuilder:validation:MinItems=1
+	// +kubebuilder:validation:items:MinLength=1
+	// +listType=set
+	// +optional
+	Files []string `json:"files,omitempty"`
+}
+
+// CustomHTTPConfig defines HTTP configuration for each request.
+type CustomHTTPConfig struct {
+	// Custom HTTP headers to be sent along with each request.
+	// Headers that are set by Prometheus itself can't be overwritten.
+	//
+	// It requires Prometheus >= v2.55.0.
+	// +mapType:=atomic
+	// +optional
+	HTTPHeaders map[string]HTTPHeader `json:"httpHeaders,omitempty"`
```

Comment:
(suggestion) we've experimented in the past that map[] structs can have edge cases with validation markers. It would be better to use named subobjects (https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#lists-of-named-subobjects-preferred-over-maps) which will also support more fine-grained patching.

````
httpHeaders:
- name: X-Foo
  secretRefs:
  - name: my-secret
     key: key1
  - name: my-secret
     key: key2
- name: X-Bar
  secretRefs:
  - name: my-secret
     key: key3
````

Reply:
Thanks @simonpasquier 
I got it.
Change map[name] to subobjects name. Please re-check it